pkg/icingadb/history/sync.go

```diff
@@ -131,6 +137,24 @@ func (s Sync) readFromRedis(ctx context.Context, key string, output chan<- redis
 	}
 }
 
+// stats forwards items in->out and counts them (as they appear) into icingaredis.Stats.History.
+func (s Sync) stats(ctx context.Context, in <-chan redis.XMessage, out chan<- redis.XMessage) error {
```

Comment:
There's already some similar code in `deleteFromRedis()` that counts for logging purposes:

https://github.com/Icinga/icingadb/blob/0bd09c4340da1722425a15812e59849b4dda8350/pkg/icingadb/history/sync.go#L161-L166

That code and the one you added should really be in the same place (a function named `stats()` definitely is a better fit for this, on the other hand `deleteFromRedis()` could just be renamed as well).